src/js/components/tabs.js

```diff
@@ -0,0 +1,220 @@
+    // If not a tab, ignore
+    if (!currentTab) return;
+
+    // Get the data-tab value
+    const currentTabValue = currentTab.getAttribute(this.tabAttribute);
+
+    // Get the associated panel
+    const currentPanel = this.element.querySelector(`[${this.panelAttribute}="${currentTabValue}"]`);
+
+    // Activate tab
+    this.activateTab(currentPanel);
+
+  }
+
+  /**
+   * This function is used to handle all keydown events on the document. It
+   * sets up handling for arrows, end, and home.
+   * @param {Event} event
+   */
+  _handleKeydown(event) {
+    const currentTab = event.target.closest(this.tabSelector);
+    // If not a tab, ignore
+    if (!currentTab) return;
+
+    // Create an array of all the focusable elements within the modal
+    const focusList = nodeListToArray(this.tabs);
+    const nextTab = focusList.indexOf(currentTab) + 1;
+    const prevTab = focusList.indexOf(currentTab) - 1;
+
+    switch (event.keyCode) {
+      case keyCodes.right || keyCodes.down: {
+
+        !focusList[nextTab] ? focusList[0].focus() : focusList[nextTab].focus();
+
+        break;
+      }
+      case keyCodes.down: {
+
+        !focusList[nextTab] ? focusList[0].focus() : focusList[nextTab].focus();
+
+        break;
+      }
+      case keyCodes.left: {
+
+        !focusList[prevTab] ? focusList[focusList.length - 1].focus() : focusList[prevTab].focus();
+
+        break;
+      }
+      case keyCodes.up: {
+
+        !focusList[prevTab] ? focusList[focusList.length - 1].focus() : focusList[prevTab].focus();
+
+        break;
+      }
+      case keyCodes.end: {
+
+        focusList[focusList.length - 1].focus();
+
+        break;
+      }
+      case keyCodes.home: {
+
+        focusList[0].focus();
+
+        break;
+      }
+      default: {
+        break;
+      }
+    }
+  }
+
+  /**
+   * This function is used to trigger the 'tabActivated' custom event,
+   * deactivate any other tab, and activate a tab/panel pair. It removes the
+   * panel's hidden attribute, adds the .rvt-tab-open class, sets the tab's
+   * aria-selected attribute to true. It also allows developers to setup a
+   * custom callback function.
+   * @param {Function} callback 
+   */
+  activateTab(tab, callback) {
+    // Trigger tabActivated custom event. This event is used to control the process of switching between tabs.
+
+    const activationEvent = dispatchCustomEvent(
+      'tabActivated',
+      tab,
+      {
+        id: tab.dataset.tabPanel,
+      }
+    );
+
+    if (!activationEvent) return;
+
+    const trigger = this.element.querySelector(`[${this.tabAttribute}="${tab.dataset.tabPanel}"]`);
+
+    this.panels.forEach(panel => {
+      if (panel.getAttribute(this.panelAttribute) !== tab.dataset.tabPanel && !panel.hasAttribute('hidden')) {
+        this.deactivateTab(panel);
+      }
+    });
+
+    // Activate the appropriate tab/panel pair
+    tab.removeAttribute('hidden');
+    tab.classList.add('rvt-tab-open');
```

Comment:
As far as I can tell this `rvt-tab-open` CSS class  isn't used for anything and it didn't exist in the old version. I'd say we can get rid of that.